services/QuillLMS/spec/models/unit_template_spec.rb

```diff
@@ -204,6 +204,91 @@
     end
   end
 
+  describe '#student_counts_for_previously_assigned_activity' do
+    let!(:student_one) { create(:student) }
+    let!(:student_two) { create(:student) }
+    let!(:student_three) { create(:student) }
+    let!(:student_four) { create(:student) }
+    let!(:classroom_one) { create(:classroom, students: [student_one, student_two, student_three]) }
+    let!(:classroom_two) { create(:classroom, students: [student_four]) }
+    let!(:unit_one) { create(:unit) }
+    let!(:unit_two) { create(:unit) }
+    let!(:classroom_unit_one) { create(:classroom_unit, unit: unit_one, classroom: classroom_one, assigned_student_ids: classroom_one.students.ids) }
+    let!(:classroom_unit_two) { create(:classroom_unit, unit: unit_two, classroom: classroom_two, assigned_student_ids: classroom_two.students.ids) }
+
+    it 'returns the expected total and assigned student counts' do
+      classrooms = [classroom_one, classroom_two]
+      unit_one_counts = UnitTemplate.student_counts_for_previously_assigned_activity(unit_one, unit_one.classrooms)
+      unit_two_counts = UnitTemplate.student_counts_for_previously_assigned_activity(unit_two, unit_two.classrooms)
+      expect(unit_one_counts).to eq([
+        {
+          total_student_count: classroom_one.students.length,
+          assigned_student_count: classroom_unit_one.assigned_student_ids.length
+        }
+      ])
+      expect(unit_two_counts).to eq([
+        {
+          total_student_count: classroom_two.students.length,
+          assigned_student_count: classroom_unit_two.assigned_student_ids.length
+        }
+      ])
+    end
+  end
+
+  describe '#previously_assigned_activity_data' do
+    let!(:current_user) { create(:teacher_with_a_couple_classrooms_with_one_student_each) }
```

Comment:
I like to put a subject at the top of specs to describe overall what is being tested: 

So in line 239: `subject { UnitTemplate.previously_assigned_activity_data(activity_ids, current_user) }`